src/manifolds/Euclidean.jl

```diff
@@ -122,13 +127,15 @@ manifold `M`, i.e. for vectors it's just the norm of the difference, for matrice
 and higher order arrays, the matrix and ternsor Frobenius norm, respectively.
 """
 distance(::Euclidean, p, q) = norm(p .- q)
+distance(::Euclidean{Tuple{}}, p::T, q::T) where {T<:Number} = abs(p - q)
```

Comment:
```suggestion
distance(::Euclidean{Tuple{}}, p::Number, q::Number) = abs(p - q)
```
We don't need to require both `p` and `q` to have the same type.

Reply:
Should I remove the `T` everywhere and write `::Number` instead? For me writing the `T<:Number` is maybe an old habit, where that was beneficial for some reason, I don't remember completely; you're right, these days it would mainly enforce them to have the same type.

Reply:
Yes, I think you should only prefer `::T`... `where T<:Number` over `::Number` when you actually need to use `T` in body of that method.

Reply:
Yes, thanks. I hopefully just unified that.